src/realm/object-store/sync/sync_manager.cpp

```diff
@@ -473,6 +473,8 @@ static std::string string_from_partition(const std::string& partition)
                 return util::format("s_%1", static_cast<std::string>(partition_value));
             case bson::Bson::Type::ObjectId:
                 return util::format("o_%1", static_cast<ObjectId>(partition_value).to_string());
+            case bson::Bson::Type::Uuid:
+                return util::format("o_%1", static_cast<UUID>(partition_value).to_string());
```

Comment:
This is an arbitrary choice, but I'd suggest "u_..." to differentiate from object id. It doesn't matter except that it makes the file name readable for the type and value of the partition, and having the type there could help when debugging something. 

Please add the UUID type to the error message in the exception below as well for the list of supported partition types.

Reply:
Of course ...  Fixed